File: class/order app/app/src/main/java/com/example/uber/MainActivity.java
package com.example.uber;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        if (requestCode == 314){
            if (resultCode == 101){
                Bundle orderBundel = data.getExtras();
                String drink = orderBundel.getString("drink");
                String sugar = orderBundel.getString("sugar");
                String ice = orderBundel.getString("ice");

                TextView tv_order = findViewById(R.id.tv_order);
                tv_order.setText("\n飲料：" + drink + "\n甜度：" + sugar + "\n冰塊：" + ice);
            }
        }
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button btn_order = findViewById(R.id.btn_order);

        // Create Intent
        Intent orderIntent = new Intent(this, MainActivity2.class);
        int ORDER_REQUEST = 314;

        btn_order.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivityForResult(orderIntent, ORDER_REQUEST);
            }
        });

    }
}

File: homework/KT02/ch04-orderSystem/order_system/app/src/main/java/com/example/order_system/MainActivity.kt
package com.example.order_system

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        if(requestCode == 314){
            if(resultCode == 90){
                var drink = data?.extras?.getString("drink") //取得bundle內key為drink的字串資料
                var ice = data?.extras?.getString("ice")     //取得bundle內key為ice的字串資料
                var sugar = data?.extras?.getString("sugar") //取得bundle內key為sugar的字串資料

                tv_order.text = "\n\n" + drink + "\n" + sugar + "\n" + ice  //將點餐結果顯示出來
            }
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        btn_order.setOnClickListener {
            //宣告intent，從目前的Activity指向Main2Activity
            var intent = Intent(this, Main2Activity::class.java)
            startActivityForResult(intent, 314) //要求傳回值及設定requestCode
        }
    }
}
